
LAB 4:Explore a real dataset (IPL / Zomato / COVID India) – answer 10 questions with Pandas

In [6]:
import pandas as pd

data = {
    "Team": ["CSK", "MI", "RCB", "KKR", "CSK", "MI", "RCB", "CSK", "KKR", "MI"],
    "Opponent": ["MI", "RCB", "KKR", "CSK", "RCB", "KKR", "CSK", "KKR", "MI", "CSK"],
    "Runs": [180, 200, 175, 160, 190, 210, 185, 170, 195, 205],
    "Wickets": [5, 4, 6, 7, 3, 5, 4, 8, 2, 6],
    "Winner": ["CSK", "MI", "RCB", "KKR", "CSK", "MI", "RCB", "KKR", "KKR", "MI"],
    "Player_of_Match": ["Dhoni", "Rohit", "Kohli", "Russell", "Dhoni",
                        "Rohit", "Kohli", "Russell", "Russell", "Rohit"]
}

df = pd.DataFrame(data)

print("Q1:", len(df))
print("Q2:", df["Winner"].value_counts().idxmax())
print("Q3:")
print(df["Winner"].value_counts().head(3))
print("Q4:", df["Runs"].max())
print("Q5:", df["Runs"].min())
print("Q6:", df["Runs"].mean())
print("Q7:", df["Player_of_Match"].value_counts().idxmax())
print("Q8:", df.groupby("Team")["Runs"].sum().idxmax())
print("Q9:", df["Wickets"].max())
print("Q10:", len(df[df["Runs"] > 190]))

Q1: 10
Q2: MI
Q3:
Winner
MI     3
KKR    3
CSK    2
Name: count, dtype: int64
Q4: 210
Q5: 160
Q6: 187.0
Q7: Rohit
Q8: MI
Q9: 8
Q10: 4


Questions and Answers

Q1. How many matches are present in the dataset?
Answer: 10

Q2. Which team won the most matches?
Answer: MI

Q3. What are the top winning teams?
Answer: MI (3), KKR (3), CSK (2)

Q4. What is the highest team score?
Answer: 210

Q5. What is the lowest team score?
Answer: 160

Q6. What is the average score of all teams?
Answer: 187

Q7. Who won the most Player of the Match awards?
Answer: Rohit and Russell (3 each)

Q8. Which team scored the highest total runs?
Answer: MI

Q9. What is the highest number of wickets taken in a match?
Answer: 8

Q10. How many teams scored more than 190 runs?
Answer: 4